### PR TITLE
Add join, escape_string, and unescape_string

### LINKED ISSUE #272: Add join with first argument as IOBuffer() ?

### PR DESCRIPTION
**Pull request by TotalVerb**
Closes #272. See https://github.com/JuliaLang/julia/pull/16603.

### CODE REVIEW

**Comment by tkelman** [comment]
LGTM, thanks

File: README.md [hunk 1]
```diff
@@ -203,6 +203,8 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `fieldoffsets` has been deprecated in favor of `fieldoffset`[#14777](https://github.com/JuliaLang/julia/pull/14777).
 
+* `print_escaped` is now another method on `escape_string`, `print_unescaped` a method on `unescape_string`, and `print_joined` a method on `join`. [#16603](https://github.com/JuliaLang/julia/pull/16603)
```
Comment (stevengj) [inline README.md]: s/method on/method of/

**Comment by musm** [comment]
thanks!

File: README.md (hunk 1 above)
Comment (TotalVerb) [inline README.md]: :+1:

**Comment by musm** [comment]
tag?

**Comment by TotalVerb** [comment]
@musm Sure, see https://github.com/JuliaLang/METADATA.jl/pull/6054.

**Pull request merged**